eng/testing/linker/project.csproj.template

```diff
@@ -14,17 +14,7 @@
     <NETCoreAppMaximumVersion>{NetCoreAppMaximumVersion}</NETCoreAppMaximumVersion>
     <MicrosoftNETCoreAppVersion>{MicrosoftNETCoreAppVersion}</MicrosoftNETCoreAppVersion>
 
-    <!-- These can be removed once we get an SDK with https://github.com/mono/linker/pull/1385. -->
-    <LinkerNoWarn>IL2026</LinkerNoWarn>
```

Comment:
I was kind of thinking some of our tests are going to be calling methods that are `RUC`, for example the Json tests all call `JsonSerializer`.

What happens to those warnings? Are they emitted during the tests, but just ignored?

Reply:
No, these warnings in the template where just overkill as these projects where already importing illink.targets which already defined some of the no warns defined here, so that is why we are not getting warnings on the tests. This PR is mostly about cleaning out the trimming infrastructure, and then we can do another PR that will clean up some of the nowarns over in illink.targets that are no longer required.

Reply:
> as these projects where already importing illink.targets 

They do? Where? I'm not seeing it.

Reply:
Sorry I said illink.targets, I meant Microsoft.NET.ILLink.targets which suppresses most of those by default unless you set SuppressTrimAnalysisWarnings to false.

Here it suppresses the RUC one: https://github.com/dotnet/sdk/blob/0de32655abbe4bb9d95d35d624ee22b6e6cad547/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets#L58

And here it suppresses the rest: https://github.com/dotnet/sdk/blob/0de32655abbe4bb9d95d35d624ee22b6e6cad547/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets#L206-L238

Reply:
Those will be removed with @sbomer's change: https://github.com/dotnet/sdk/pull/17676. Which will then break our tests when we take that new SDK.

Reply:
Will do